src/lib/utils/hooks/useGetInfoFromIds.ts

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useState } from 'react'
+import { EntityHeader, Reference, ReferenceList } from '../synapseTypes'
+import { getEntityHeader } from '../SynapseClient'
+import { getUserProfileWithProfilePicAttached } from '../functions/getUserData'
+import { UserProfile } from '../synapseTypes'
+import { SynapseConstants } from '..'
+import { without, chunk } from 'lodash-es'
+
+export type HookType = 'ENTITY_HEADER' | 'USER_PROFILE'
+export type UseGetInfoFromIdsProps = {
+  ids: string[]
+  token?: string
+  type: HookType
+}
+
+type LookupRequestType = string | Reference
+
+const UserProfileTemplate: UserProfile = {
+  ownerId: '', // A foreign key to the ID of the 'principal' object for the user
+  firstName: 'Unknown', // This person's given name (forename)
+  lastName: 'Unknown', // This person's family name (surname)
+  userName: 'Unknown', // A name chosen by the user that uniquely identifies them
+}
+
+const entityHeaderTemplate: EntityHeader = {
+  name: 'Unknown', //	The name of the entity
+  id: 'unknown', //	The id of the entity
+  type: 'unknown', //	The type of the entity
+  versionNumber: 0, //	The version number of the entity
+  versionLabel: 'placeholder', //	The user defined version label of the entity
+  benefactorId: 0, //	The ID of the entity that this Entity's ACL is inherited from.
+  createdOn: 'null', //	The date this entity was created.
+  modifiedOn: 'null', //	The date this entity was last modified.
+  createdBy: 'null', //	The ID of the user that created this entity.
+  modifiedBy: 'null', //	The ID of the user that last modified this entity.
+}
+
+const getEntityHeaderItems = async (
+  lookupList: ReferenceList,
+  token: string | undefined,
+): Promise<EntityHeader[]> => {
+  const newData = await getEntityHeader(lookupList as ReferenceList, token)
+  const notFound = lookupList.filter(
+    item =>
+      newData.results.map(item => item.id).indexOf(item.targetId) == -1,
+  )
+  const notFoundPlaceholders = notFound.map(item => ({
```

Comment:
One downside of this notFoundPlaceholders approach is that if the user logs in then any stored data won't update with the new token, if we could know that the token specifically changed in the useEffect then we could call this function all over again, but I don't believe that information is surfaced from useEffect. This problem isn't unique to this hook (or the previous implementation for that matter), any of our hooks which store data over multiple updates will have stale information from the token getting updated, I think this issue will have to be put on the back burner.

Reply:
my understanding that hook will rerun because the token is in the
[dependency array](https://github.com/Sage-Bionetworks/Synapse-React-Client/pull/418/files/e2e08bb1a90550ef36763493be4c21238b6df0ef#diff-7b41d0b46439218454ac9ff0541f5830R123)
You don't think so?

Reply:
I agree that it will re-run with the token change and any new data coming in will be fine, but the existing data won't update. Since dummy objects are stored for all the synids that don't have a return item -- the ones the user doesn't have access too -- when the user log's in those won't update.

e.g.
1. If useGetInfoFromIds is called without a token (so logged out) and a list of entity header ids, let's say `[syn123, syn456, syn789]` the first two syn ids get a return mapped to them `[{name: 'zzzz', id: 'syn123'}, {name: 'xxxx', id: 'syn789'}]` and the last one `syn789` the user didn't have permission to view when signed out, so it has a faux entry stored `[{name: 'No Header', id: 'syn789'}]`

2. Now the user sign's in with their updated token (which should give them access to `syn789`) - the useEffect only call's get entity header if the list of incoming synIds isn't contained in the stored list of entity headers. In the case of an updated token, the incoming synIds are no different, only the token changed, so now `syn789` is stored as this faux entry and the data is stale.

Reply:
I see what you are saying! I wonder how easy it would be to set up the test case.  I am thinking (a hunch;) that in practical terms changing the token would cause the rerender of the parent component which would it turn cause the rerender of the hook. May be I can test that.